plugin/trino-apache-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControl.java

```diff
@@ -0,0 +1,1097 @@
+        String transformer = null;
+
+        if (maskTypeDef != null) {
+            transformer = maskTypeDef.getTransformer();
+        }
+
+        if (RangerPolicy.MASK_TYPE_NULL.equalsIgnoreCase(maskType)) {
+            transformer = "NULL";
+        }
+        else if (RangerPolicy.MASK_TYPE_CUSTOM.equalsIgnoreCase(maskType)) {
+            String maskedValue = result.getMaskedValue();
+
+            transformer = requireNonNullElse(maskedValue, "NULL");
+        }
+
+        if (transformer == null) {
+            return Optional.empty();
+        }
+        else {
+            transformer = transformer.replace("{col}", columnName).replace("{type}", type.getDisplayName());
+
+            return Optional.of(ViewExpression.builder().identity(context.getIdentity().getUser()).catalog(tableName.getCatalogName()).schema(tableName.getSchemaTableName().getSchemaName()).expression(transformer).build());
+        }
+    }
+
+    @Override
+    public Iterable<EventListener> getEventListeners()
+    {
+        return ImmutableList.of(eventListener);
+    }
+
+    @Override
+    public void shutdown()
+    {
+        rangerPlugin.cleanup();
+    }
+
+    private RangerAccessResult getDataMaskResult(RangerTrinoAccessRequest request)
+    {
+        return rangerPlugin.evalDataMaskPolicies(request, null);
+    }
+
+    private RangerAccessResult getRowFilterResult(RangerTrinoAccessRequest request)
+    {
+        return rangerPlugin.evalRowFilterPolicies(request, null);
+    }
+
+    private boolean isDataMaskEnabled(RangerAccessResult result)
+    {
+        checkArgument(result != null);
+
+        return result.isMaskEnabled();
+    }
+
+    private boolean isRowFilterEnabled(RangerAccessResult result)
+    {
+        checkArgument(result != null);
+
+        return result.isRowFilterEnabled();
+    }
+
+    private RangerTrinoAccessRequest createAccessRequest(RangerTrinoResource resource, SystemSecurityContext context, RangerTrinoAccessType accessType, String action)
+    {
+        Set<String> userGroups = context.getIdentity().getGroups();
+
+        return new RangerTrinoAccessRequest(resource, context.getIdentity().getUser(), userGroups, getQueryTime(context), getClientAddress(context), getClientType(context), getQueryText(context), accessType, action);
+    }
+
+    private RangerTrinoAccessRequest createAccessRequest(RangerTrinoResource resource, Identity identity, QueryId queryId, RangerTrinoAccessType accessType, String action)
+    {
+        Set<String> userGroups = identity.getGroups();
+
+        return new RangerTrinoAccessRequest(resource, identity.getUser(), userGroups, getQueryTime(queryId), getClientAddress(queryId), getClientType(queryId), getQueryText(queryId), accessType, action);
+    }
+
+    private String getClientAddress(QueryId queryId)
+    {
+        if (queryId != null) {
+            return eventListener.getClientAddress(queryId.getId()).orElse(null);
+        }
+        else {
+            return null;
```

Comment:
Please do not return `null`. Use `Optional` instead. Same below.

Considering the comment above, it looks like the implementation of this method will get simplified, so please consider inlininng.